test/utils/pebble.go

```diff
@@ -0,0 +1,142 @@
+package testutils
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"net/http"
+	"os"
+	"os/exec"
+	"runtime"
+
+	"github.com/go-logr/logr"
+	"github.com/letsencrypt/pebble/v2/ca"
+	"github.com/letsencrypt/pebble/v2/cmd"
+	"github.com/letsencrypt/pebble/v2/db"
+	"github.com/letsencrypt/pebble/v2/va"
+	"github.com/letsencrypt/pebble/v2/wfe"
+)
+
+// The default values for the Pebble config have been taken from: https://github.com/letsencrypt/pebble/blob/main/test/config/pebble-config.json
+const (
+	listenAddress             = "localhost:14000"
+	ocspResponderURL          = ""
+	alternateRoots            = 0
+	chainLength               = 1
+	certificateValidityPeriod = 0
+	httpPort                  = 5002
+	tlsPort                   = 5001
+	strict                    = true
+	customResolverAddr        = ""
+	requireEAB                = false
+	retryAfterAuthz           = 3
+	retryAfterOrder           = 5
+)
+
+// RunPebble runs a pebble server with the given configuration.
+// The code is copied, shortened, and adapted from: https://github.com/letsencrypt/pebble/blob/main/cmd/pebble/main.go
+func RunPebble(logr logr.Logger) (certificatePath, directoryAddress string, err error) {
+	// We don't want to go through DNS-01 challenges in the integration tests as we would have to spin up a local, authoritative DNS server.
+	// Setting the environment variable PEBBLE_VA_ALWAYS_VALID to 1 makes the Pebble server always return a valid response for the validation authority.
+	// Testing the DNS-01 challenge is covered by the functional E2E tests.
+	// See the Pebble documentation: https://github.com/letsencrypt/pebble#skipping-validation
+	err = os.Setenv("PEBBLE_VA_ALWAYS_VALID", "1")
+	if err != nil {
+		return "", "", fmt.Errorf("failed to set environment variable: %v", err)
+	}
+
+	certificatePath, privateKeyPath, err := generateCertificate()
+	if err != nil {
+		return "", "", err
+	}
+
+	log := NewLogBridge(logr)
+
+	database := db.NewMemoryStore()
+	certificateAuthority := ca.New(log, database, ocspResponderURL, alternateRoots, chainLength, certificateValidityPeriod)
+	validationAuthority := va.New(log, httpPort, tlsPort, strict, customResolverAddr, database)
+
+	wfeImpl := wfe.New(log, database, validationAuthority, certificateAuthority, strict, requireEAB, retryAfterAuthz, retryAfterOrder)
+	muxHandler := wfeImpl.Handler()
+
+	directoryAddress = fmt.Sprintf("https://%s%s", listenAddress, wfe.DirectoryPath)
+
+	log.Printf("Listening on: %s", listenAddress)
+	log.Printf("ACME directory available at: %s",
+		directoryAddress)
+
+	go func() {
+		err := http.ListenAndServeTLS(
+			listenAddress,
+			certificatePath,
+			privateKeyPath,
+			muxHandler)
```

Comment:
Resolve gosec issue
```suggestion
		server := &http.Server{
			Addr:    listenAddress,
			Handler: muxHandler,
			// Set timeouts to avoid Slowloris attacks and other issues
			ReadTimeout:  10 * time.Second,
			WriteTimeout: 10 * time.Second,
			IdleTimeout:  15 * time.Second,
		}
		err := server.ListenAndServeTLS(certificatePath, privateKeyPath)
```